### ISSUE TITLE
Underextrusion after overhangs because smoothing flow/speed is not working

### ISSUE THREAD

**Issue by d-schmidt** [SoftFever/OrcaSlicer#7804]
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### OrcaSlicer Version

2.2.0

### Operating System (OS)

Windows

### OS Version

Windows 10

### Printer

Creality K1C

### How to reproduce

Enable fast speeds and smoothing and have slow overhangs

### Actual results

Underextrusion:
![grafik](https://github.com/user-attachments/assets/746b561d-ae8f-494f-bb48-80c2dd6547cc)

### Expected results

It takes a moment to accelerate, which seems to work at some places but these corners cause it to switch to full speed immediatly

### Project file & Debug log uploads

[worx battery holder-orange.zip](https://github.com/user-attachments/files/18215678/worx.battery.holder-orange.zip)

### Checklist of files to include

- [ ] Log file
- [X] Project file

**Comment by d-schmidt** [comment]
It looks like the layer time limit is exacerbating the problem. It moves faster on layers where it has to slow down for overhangs.
The speed itself isn't a problem, the printer and filament can keep up. given enough time to get the flow up again.

![grafik](https://github.com/user-attachments/assets/2cd63be1-50c8-4128-af21-618ad023f07e)

**Comment by d-schmidt** [comment]
Also somethings seems funky with the Smoothing segment length:

1:
![grafik](https://github.com/user-attachments/assets/8d4ee9ca-ddf9-4df5-9d9d-d6af77abda58)

2:
![grafik](https://github.com/user-attachments/assets/3269df73-f652-4e1f-8eb3-bec801d13976)

3:
![grafik](https://github.com/user-attachments/assets/9d09d3b2-fa16-46c6-bf68-1e29d5cb4471)

4:
![grafik](https://github.com/user-attachments/assets/3aa27a0f-46e5-4d34-8e84-01c74897aa3f)

5:
![grafik](https://github.com/user-attachments/assets/c413e200-de54-431f-b2f2-b2ecf8299270)

**Comment by Otarifpv** [comment]
I'm also suffering from this problem, I've been trying to deal with the smoothing, but I can only hide it, I can't get it to print correctly, I also see that there are layers in which the printer for some reason triggers the layer time, and cooling is not everything as regular as I would like, no matter how I configure it I never get what I expected, I also think that some bridges fail for this same reason, the area where it is anchored on the perimeters used a speed that was too high and this would cause the bridge not to anchor correctly.

excuse my english

**Comment by d-schmidt** [comment]
Nobody got any idea how smoothing, layer time and overhang mess each other up?

Also for asthetic reasons the layertime shouldn't switch this hard. A 180mm/s surface will never look the same as the 80mm/s layer below.

**Comment by fridaykillme** [comment]
Im not sure what i have same problem, but very similar. FF ad5m pro. Early i have not underextrusion after overhangs/turns.

**Comment by igiannakas** [comment]
apply extrusion rate smoothing - latest nightly also has an option to limit this to external perimeters only.

**Comment by d-schmidt** [comment]
> apply extrusion rate smoothing - latest nightly also has an option to limit this to external perimeters only.

@igiannakas the last screenshots are with extrusion rate smoothing. It is bugged. I added a screenshot for each (1-5) extrusion rate smoothing length

**Comment by igiannakas** [comment]
Set segment length to 1, ERS speed to 50 and only on outer walls.  Segment size is how big the length of the step in mm where the slowdown happens. ERS speed is the rate of volumetric change that is allowed.

![Image](https://github.com/user-attachments/assets/1c11288a-4c61-4477-b7a5-4df237f5434c)

You can go below 1 if needed, which will give you a smoother transition at the risk of stuttering if the printer's control board cannot keep up with the segment size. Eg below is 0.5mm length.

![Image](https://github.com/user-attachments/assets/2538f87b-007e-4746-b02c-3f60e093860c)

You shouldn't need to use it through. What you are seeing is the extruder struggling to keep up with speed changes. Is your PA tuned accurately?

**Comment by igiannakas** [comment]
> Nobody got any idea how smoothing, layer time and overhang mess each other up?
> 
> Also for asthetic reasons the layertime shouldn't switch this hard. A 180mm/s surface will never look the same as the 80mm/s layer below.

I tried introducing layer time smoothing but the effects were marginally worse actually.https://github.com/SoftFever/OrcaSlicer/pull/8107  The sharp transition was spread over multiple layers. Suggestion is to keep constant speeds on outer walls which helps a ton with model surface appearance. this can be found in the filament settings - dont slow down outer walls.

**Comment by d-schmidt** [comment]
The nightly build still has these "artifacts" throwing off my printer.

The default 300 - 3 looks like this:
![Image](https://github.com/user-attachments/assets/f90245bf-e803-4601-be27-472524e9df71)
These can't be working as intended. Given a bit time the printer handles everything fine. But these hard magnitude speed changes are too much. 

I will test wether my printer can handle the length of 1 mm. The stable has a warning there. So far this seemse the only length working well. At least in the preview.

**Comment by igiannakas** [comment]
Did you try the settings I set above? 50 ERS and 1mm segment? The screenshots were taken with the project file and these two settings set as per the screenshot. 300 ERS may be too high for your specific speed settings.

If you want attach your latest project file, but first please try the settings I suggested.

The build I'm using is here: https://github.com/igiannakas/OrcaSlicer/actions/runs/13077675455 

It will be done compiling in around 1h. Its a few commits ahead of the nightly.

**Comment by d-schmidt** [comment]
I did try 50 ERS and 1 mm. It looks much better and I will check wether my pritner can handle the extra load. It's an K1C so I hope it works.

The screenshots are from the "default" 300-3 settings.

50-1 looks like this for mee in the latest available nigthly build:

![Image](https://github.com/user-attachments/assets/dfba8320-26d7-4dd5-9ea3-d0e04fe914ac)

**Comment by igiannakas** [comment]
Looks about right. I presume the project file you've got open right now is slightly different to the one in the start of the thread? In any case, yes these will be more gradual.

However, the printer really shouldn't need this setting - it's there to remove minor surface imperfections, but what you're getting is major under extrusion. Its supposed to correct issues like this

![Image](https://github.com/user-attachments/assets/416d3385-6544-459d-8638-0268a3cb5012) https://github.com/SoftFever/OrcaSlicer/pull/2161

Which are not as severe as the ones you're seeing. What you are seeing is caused by incorrect PA and / or an extruder issue and / or excessive drag in the filament path.

**Comment by d-schmidt** [comment]
Okay, I will try to calibrate PA for that fillament.

I just assumed it is something smoothing should help with as the print only failed at the 10->110mm/s switch at that place because the slicer wants to make up for the lost time printing the overhang. If PA should actually prevent these artifacts after magnitude speed-switches I will put some time into them.

**Comment by igiannakas** [comment]
Correct. Pa is a static value set per filament. There’s nothing to see in the preview as it’s directly sent to the printer to use in its extruder planning routines.

PA can make absolutely a huge difference in speed transitions. That’s what it’s made for :)

Suggest to have a read here: https://ellis3dp.com/Print-Tuning-Guide/articles/pressure_linear_advance/introduction.html

**Comment by d-schmidt** [comment]
You want me to keep this ticket open for the weird smoothing behavior or should I close it?

![Image](https://github.com/user-attachments/assets/d377024b-00fa-4cd6-9167-7ca636f5f73a)

**Comment by hotellonely** [comment]
Many people are having the similar issue but it looks like we don't have a definitive answer.

1. PA
PA, at least a static PA, won't be able to solve this issue. Many have tried including myself. A dynamic PA calibration for multi material or various material is tooooo time costly.

2. Extruder
Even very popular fast printing machiens like X1C, K1C are having this issue. 

3. ESR/Smooth speed
**No, even ESR cannot fully fix it. Even with a very low extrusion smoothing threshold, the under extrusion is still showing up.**

4. Turning off Slow Down for Overhang
It DOES help! But at the cost of overhang performance. 

5. Misplaced seams
Sometimes slicer would create seams on overhang area. Causing the overextrusion to layer up. Eventually causing a micro-clog.

6. **WARP**, or **environment too hot**, or not enough minimum layer time.
Quite important factor actually. Small overhangs like these tend to warp up. Part cooling in this case don't help much, you need the environment to be cool. At least that is how I mitigated much of my exactly the same problem.

I don't know the exact reason, but I've explored many options, so here's my two cent.

**Comment by sebastianha** [comment]
Hi,
is there any "real" solution for this problem? I recently had the same problem and now trying to fix it by tuning max flow rate, overhang speed and extrusion rate smoothing. But nothing is really satisfying.

**Comment by igiannakas** [comment]
> Hi, is there any "real" solution for this problem? I recently had the same problem and now trying to fix it by tuning max flow rate, overhang speed and extrusion rate smoothing. But nothing is really satisfying.

Start by tuning your pressure advance - as good as you can get it. This by far has the biggest impact on how consistent your extrusion is before and after an overhang.

https://youtu.be/jNQYUss37pc?si=nWVps4WACTlmI6I-

**Comment by sebastianha** [comment]
I was pretty sure I already did this but I did it again for this filament and I needed to increase it a lot. I think now it's too much but definitely thanks for the hint!
